app/questionnaire/questionnaire_manager.py

```diff
@@ -125,3 +126,44 @@ def get_schema_item_by_id(self, item_id):
 
     def get_schema(self):
         return self._schema
+
+    def add_answer(self, block, question, post_data, answer_store):
+        if self.state is None:
+            self.process_incoming_answers(block, post_data)
+
+        question_schema = self._schema.get_item_by_id(question)
+        question_state = self.state.find_state_item(question_schema)
+
+        answer_schema = question_schema.answers[0]  # Single answer for now.
```

Comment:
I think renaming `answer_schema` to `existing_answer_schema` would make things clearer